lib/build/cpp/Dockerfile

```diff
@@ -0,0 +1,44 @@
+ARG BUILD_BASE_IMG_TAG
+FROM symbolica/build:${BUILD_BASE_IMG_TAG}
+LABEL maintainer "Symbolica <dev@symbolica.dev>"
+
+RUN apk add --no-cache linux-headers && \
+    cp -r /usr/include/* /home/me/.symbolica/build/sysroot/include/ && \
+    apk --purge del linux-headers
+
+COPY ./c++ $HOME/.symbolica/build/c++
+
+RUN apk add --no-cache --virtual build-deps git cmake ninja python3 && \
+    cd /tmp && \
+    git clone --depth 1 --branch llvmorg-12.0.1 https://github.com/llvm/llvm-project.git && \
+    cd llvm-project \
+    mkdir build && \
+    CODEGEN=1 NO_LLVM=1 CC=~/.symbolica/build/cc CXX=~/.symbolica/build/c++ \
+        cmake -G Ninja \
+        -S runtimes \
+        -B build \
+        -DCMAKE_BUILD_TYPE=MinSizeRel \
+        -DCMAKE_INSTALL_PREFIX="${HOME}/.symbolica/build/sysroot" \
+        -DLIBCXXABI_ENABLE_SHARED=OFF \
+        -DLIBCXXABI_USE_LLVM_UNWINDER=ON \
+        -DLIBCXX_CXX_ABI=libcxxabi \
+        -DLIBCXX_ENABLE_EXPERIMENTAL_LIBRARY=OFF \
+        -DLIBCXX_ENABLE_SHARED=OFF \
+        -DLIBCXX_HAS_GCC_S_LIB=OFF \
+        -DLIBCXX_HAS_MUSL_LIBC=ON \
+        -DLIBUNWIND_ENABLE_SHARED=OFF \
+        -DLLVM_ENABLE_RUNTIMES="libcxx;libcxxabi;libunwind" && \
+    CODEGEN=1 ninja -C build cxx cxxabi unwind && \
+    # # TODO: Enable testing once it's working
+    # # ninja -C build check-cxx check-cxxabi check-unwind && \
+    ninja -C build install-cxx install-cxxabi install-unwind && \
+    mkdir -p ~/.symbolica/build/c++ && \
+    find build/libunwind -name "*.o.bc" -print0 | xargs -0 ~/.symbolica/build/llvm-link -o ~/.symbolica/build/c++/libunwind.bc && \
```

Comment:
@iscottb122 I've literally just slung these llvm link lines in. They're currently not working for me, but I thought I'd put them here in case you want to take over from this point so you can make some progress based off of the rest of it.

Specifically, when I commented out the lines after `ninja -C build install` and then tried to `llvm-link` libunwind, I got the following error:

```sh
llvm-link-12: build/libunwind/src/CMakeFiles/unwind_static.dir/UnwindRegistersRestore.S.o.bc:1:1: error: expected top-level entity
# 1 "/tmp/llvm-project/libunwind/src/UnwindRegistersRestore.S"
^
llvm-link-12: error:  loading file 'build/libunwind/src/CMakeFiles/unwind_static.dir/UnwindRegistersRestore.S.o.bc'
```

I've literally just done this so apologies for not taking any time to investigate this one of my own.